src/configuration/remapper.ts

```diff
@@ -150,36 +150,39 @@ export class Remapper implements IRemapper {
     vimState.keyHistory = vimState.keyHistory.slice(0, -numCharsToRemove);
 
     if (remapping.after) {
+      await modeHandler.handleMultipleKeyEvents(remapping.after);
```

Comment:
Can you explain why this is only being done once now?

Reply:
All the actions already handle the case when there is a count. Some of them have a special handle that applies the entire counts in one go others have the `for` loop on the `BaseAction.execCount`. This is what the modeHandler always calls even if `recordedState.count` is 0. Previously on remaps we were setting the count to 0 and then using a `for` loop to call the remaps 'count' number of times but that means that the actions would always be run with `execCount(0)` so if you had a remap to `2V` instead of selecting two lines like the `execCount(2)` would've done we are call `V` twice meaning we go to VisualLine mode and then we leave VisualLine mode.
This might be from an older version were the actions didn't handle count properly, but now they do so this was unnecessary. If you map `v -> V` and you pressed `2v` it would do nothing because it would make `recordedState.count=0` and handled 'VV', but what you want is `2v` to become `2V` after remap, and when you get to the `v` remap the '2' was already handled and `recordedState.count=2` so when handling the remap all you need to do is handling the `V` without changing the count.

Reply:
Bleh. I remember taking a look at this stuff a while back and thinking it needs a refactor. Thanks for the explanation!

Reply:
Also in the situation where you did `3i=jk` with `jk -> <Esc>` in insert mode this would result in `3i=<Esc><Esc><Esc>`. But when handling the first <Esc> the `recordedState.count` was set to 0 so the `CommandEscInsertMode` which is responsible to handle the `[count]i` to repeat the insertions would see that `recordedState.count` was 0 and it wouldn't repeat like it should.

Reply:
> 
> 
> Bleh. I remember taking a look at this stuff a while back and thinking it needs a refactor. Thanks for the explanation!

If you ever need any help understanding something from the `handleKeyEvent`, `handleMultipleKeyEvents`, `handleKeyEventHelper` or even some parts from `runAction` from modeHandler or anything from `remapper` you can always ask me since I think I'm starting to understand it all a little bit better now. 😄